src/dashboard/src/main/management/commands/rebuild_transfer_backlog.py

```diff
@@ -166,22 +188,70 @@ def populate_data(self, es_client, transfer_backlog_dir):
                 self.info(
                     "Importing self-describing transfer {}.".format(transfer_uuid)
                 )
-                size = get_bag_size(bag, str(transfer_dir))
+                size = am.get_bag_size(bag, str(transfer_dir))
                 _import_self_describing_transfer(
                     self, es_client, self.stdout, transfer_dir, transfer_uuid, size
                 )
             else:
                 self.info("Rebuilding known transfer {}.".format(transfer_uuid))
                 if bag:
-                    size = get_bag_size(bag, str(transfer_dir))
+                    size = am.get_bag_size(bag, str(transfer_dir))
                 else:
-                    size = walk_dir(str(transfer_dir))
+                    size = am.walk_dir(str(transfer_dir))
                 _import_pipeline_dependant_transfer(
                     self, es_client, self.stdout, transfer_dir, transfer_uuid, size
                 )
             processed += 1
         self.success("{} transfers indexed!".format(processed))
 
+    def populate_data_from_storage_service(self, es_client, pipeline_uuid):
+        """Populate data in search indices and/or database from files temporarily downloaded from Storage Service.
+
+        :param es_client: Elasticsearch client.
+        :param pipeline_uuid: UUID of origin pipeline for transfers to reindex.
+
+        :returns: None
+        """
+        transfers = storageService.get_file_info(package_type="Transfer")
+        filtered_transfers = am.filter_packages(transfers, pipeline_uuid=pipeline_uuid)
+        processed = 0
+        for transfer in filtered_transfers:
+            transfer_uuid = transfer["uuid"]
+            temp_backlog_dir = tempfile.mkdtemp()
+            try:
+                local_package = storageService.download_package(
+                    transfer_uuid, temp_backlog_dir
+                )
+            except storageService.Error:
+                self.error(
+                    "Transfer {} not indexed. Unable to download from Storage Service.".format(
+                        transfer_uuid
+                    )
+                )
+                continue
+            # Transfers are downloaded as .tar files, so we extract files before indexing.
+            extract_package(local_package, temp_backlog_dir)
```

Comment:
This stills assumes extraction worked, correct?

I was wondering what would happen if the `subprocess.call` fails with a return code other than `0`. Should we at least log that?

Reply:
Yes, good point! I had taken a more indirect route of throwing an error when the directory expected to be extracted wasn't present, but you're right that it would be better to explicitly check the return code. I will add that before merging.